File: C++/reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        sort(deck.begin(), deck.end());
        
        queue<int> index;
        int sz = deck.size();
        for (int i = 0; i < sz; i++) {
            index.push(i);
        }
        
        vector<int> ans(sz);
        for (int card: deck) {
            ans[index.front()] = card;
            index.pop();
            if (!index.empty()) {
                index.push(index.front());
                index.pop();
            }
        }
        
        return ans;
    }
};

File: Go/reveal-cards-in-increasing-order.go
func deckRevealedIncreasing(deck []int) []int {
    // sort cards in desired order
    slices.Sort(deck)

    // make a queue of indices to simulate drawing
    queue := make([]int, len(deck))
    for i := range queue {
        queue[i] = i
    }

    // cards will be placed on their desired position
    result := make([]int, len(deck))

    for _, card := range deck {
        // simulate drawing the "show" card
        result[queue[0]] = card
        queue = queue[1:]
        if len(queue) > 0 {
            // simulate drawing the "skip" card and placing it on the back of the queue/deck
            queue = append(queue[1:], queue[0])
        }
    }

    return result
}